src/types.rs

```diff
@@ -139,15 +142,27 @@ impl Request {
             method: req.method().clone(),
             params: HashMap::new(),
             body,
+            path: req.path().to_string(),
+            host: req.connection_info().host().to_string(),
+            scheme: req.connection_info().scheme().to_string(),
```

Comment:
Thanks for your comment! Now, this looks awesome ;)

Reply:
I am glad you liked the comments @r3b-fish 😄 .

I have one more follow up comment before we can merge this. We can add one more optimization and use the `intern!` macro from py03.

So the code will look something like this:

```rust
struct URL {
    method: String;
    path: String;
...
}

impl URL {
    fn new(method: &str, path: &str, ...) -> Self {
        Self()        
    }

    pub fn to_object(&self, py: Python<'_>) -> PyResult<PyObject> {
        let dict = PyDict::new(py);
        let method = self.method
        dict.set_item(intern!(py, "method"), self.method)?;
        dict.set_item(intern!(py, "path"), self.method)?;
        Ok(dict.into())
            
}
}

```


Using this will help us create this string without reallocations.

You can read more about it here - https://pyo3.rs/main/doc/pyo3/macro.intern.html

Reply:
Thanks for you help. I'm not sure about the "method". Can you check my code again, please.

Reply:
@r3b-fish , `method` was a typo from my end. My bad!


Great work btw! :smile: